src/piadm.sh

```diff
@@ -655,13 +675,250 @@ ispoolenabled() {
+		# Need an extra "platform" in here to satisfy the illumos
+		# load-time that looks for a unix in the boot archive.  The
+		# boot file path MUST have /platform/i86pc/kernel/amd64/unix
+		# as its trailing components.  See install_pi_CN() for the
+		# insertion of a symlink to help out.
+		echo bootfile=\"/platform-"$installstamp"/platform/i86pc/kernel/amd64/unix\"
+		echo boot_archive_name=\"/platform-"$installstamp"/i86pc/amd64/boot_archive\"
+		echo boot_archive.hash_name=\"/platform-"$installstamp"/i86pc/amd64/boot_archive.hash\"
+	} >> boot-ipxe/loader.conf
+
+	# Caller will invoke update_boot_sectors.
+}
+
+update_CN() {
+	declare -a pdirs
+
+	if [[ "$TRITON_CN" != "yes" ]]; then
+		err "The update command may only be used on a Triton Compute Node"
+	fi
+
+	piname_present_get_bootfs "ipxe" "$1"
+	cd "/${bootfs}" || fatal "Could not chdir to /$bootfs"
+
+	# First check if the backup PI is in need of update.
+	# The standard iPXE/CN deployment has exactly one platform-STAMP.
+	mapfile -t pdirs < <(ls -d platform-[0-9]*T*Z)
+	if [[ ${#pdirs[@]} -gt 1 ]]; then
+		corrupt "Multiple platform-STAMP in CN bootfs /${bootfs}/."
+	elif [[ ${#pdirs[@]} -lt 1 ]]; then
+		corrupt "No platform-STAMP in CN bootfs /${bootfs}/."
+	fi
+
+	pdir=${pdirs[0]}
+
+	pstamp=$(cat "${pdir}"/etc/version/platform)
+	if [[ "$pstamp" != "$activestamp" ]]; then
+		vecho "Updating backup PI to" "$activestamp"
+		if ! install_pi_CN "$activestamp" && \
+			[[ "$CNAPI_DEFAULT_PI" != "$activestamp" ]]; then
+			vecho "...trying" "$CNAPI_DEFAULT_PI" "instead"
+			if ! install_pi_CN "$CNAPI_DEFAULT_PI" ; then
+				/bin/rm -rf platform-"$activestamp"
+				/bin/rm -rf platform-"$CNAPI_DEFAULT_PI"
+				err "No PIs available, keeping" "$pstamp"
+			fi
+		fi
+
+		# Success, don't keep the old one!
+		# $installstamp will have new PI stamp, will inform below.
+		vecho "...success installing $installstamp"
+		/bin/rm -rf "$pdir"
+		tfile=$(mktemp)
+		# Alter loader.conf's backup PI stamp.
+		# NOTE: If loader.conf has a flag day, we really need to
+		# Do Better here.
+		cp ./boot-ipxe/loader.conf "${tfile}"
+		sed s/"${pstamp}"/"${installstamp}"/g < "${tfile}" \
+			> ./boot-ipxe/loader.conf
+		rm -f "${tfile}"
+	fi
+
+	# THEN check to see if we need to update iPXE...
+	diskipxe=$(cat etc/version/ipxe)
+	diskboot=$(cat etc/version/boot)
+	newipxe=$(cat ${TRITON_IPXE_ETC}/version/ipxe)
+	newboot=$(cat ${TRITON_IPXE_ETC}/version/boot)
+	if [[ "$diskipxe" == "$newipxe" && "$diskboot" == "$newboot" ]]; then
+		vecho "No updates needed for iPXE and its loader."
+		vecho "If you think there should be an update, run"
+		vecho "'sdcadm experimental update-gz-tools' on your"\
+			"Triton Head Node" 
+		exit 0
+	fi
+
+	[[ "$diskipxe" != "$newipxe" ]] && \
+		vecho "Updating iPXE provided by headnode (ver: $newipxe)"
+	[[ "$diskboot" != "$newboot" ]] && \
+		vecho "Updating boot loader provided by headnode (ver: $newboot)"
+	cp -f ${TRITON_IPXE_ETC}/version/* etc/version/.
```

Comment:
"`${TRITON_IPXE_ETC}"`

Reply:
See above - it's hardcoded and not environmentally overridden.